### PR TITLE
Deploy using Travis

### LINKED ISSUE #131: Automate wheels generation and pypi upload in TravisCI
As per comments in #111 and #127 we should automatically deploy wheels as part of the CI process.
@nicoddemus mentioned that we can do the same as in [`pytest-xdist`](https://github.com/pytest-dev/pytest-xdist/blob/4dc86a1dad8e1924064db32d5b9ea49f2e33c2e1/.travis.yml#L50-L59).

Ideally this is working for the next release!

### PR DESCRIPTION
**Pull request by nicoddemus**
This also uses `setuptools_scm` so we don't have to update the version manually.

Fix #131

### CODE REVIEW

**Comment by nicoddemus** [comment]
Also noticed we did not document how to do a release, added one.

File: .travis.yml [hunk 1]
```diff
@@ -30,6 +30,22 @@ matrix:
...
 
+    - stage: deploy
+      python: '3.6'
+      env:
+      install: pip install -U setuptools setuptools_scm
+      script: skip
+      deploy:
+        provider: pypi
+        user: nicoddemus
+        distributions: sdist bdist_wheel
```
[the remark this answers is not in the document]
Comment (nicoddemus) [reply .travis.yml]: Good question, but [we already have that](https://github.com/pytest-dev/pluggy/blob/master/setup.cfg#L1).

[the remark this answers is not in the document]
Comment (tgoodlet) [reply .travis.yml]: Shoot, right. It got me again.

**Comment by jezdez** [comment]
Did you also upload non-universal wheel files to PyPI for 0.6.0 retroactively? Just asking since this broke out hash-pinned requirements files and I'm trying to understand how they ended up on PyPI.

**Comment by RonnyPfannschmidt** [comment]
@jezdez whops and the link for the issue is #134

**Pull request merged**